ddsrecorder/test/blackbox/mcap/McapFileCreationTest.cpp

```diff
@@ -0,0 +1,456 @@
+        mcap_handler);
+
+    // Create and populate Participant Database
+    std::shared_ptr<core::ParticipantsDatabase> participant_database =
+            std::make_shared<core::ParticipantsDatabase>();
+
+    // Populate Participant Database
+    participant_database->add_participant(
+        dyn_participant->id(),
+        dyn_participant
+        );
+    participant_database->add_participant(
+        recorder_participant->id(),
+        recorder_participant
+        );
+
+    return std::make_unique<core::DdsPipe>(
+        allowed_topics,
+        discovery_database,
+        payload_pool,
+        participant_database,
+        thread_pool,
+        configuration.builtin_topics,
+        true
+        );
+}
+
+void create_publisher(
+        std::string topic_name,
+        uint32_t domain,
+        DataTypeKind data_type_kind)
+{
+    eprosima::fastdds::dds::DomainParticipantQos pqos;
+    pqos.name("TypeIntrospectionExample_Participant_Publisher");
+    pqos.wire_protocol().builtin.typelookup_config.use_client = false;
+    pqos.wire_protocol().builtin.typelookup_config.use_server = true;
+
+    // Create the Participant
+    eprosima::fastdds::dds::DomainParticipant* participant_ =
+            DomainParticipantFactory::get_instance()->create_participant(domain, pqos);
+
+    // Register the type
+    registerHelloWorldTypes();
+    test::dynamic_type_ = eprosima::fastrtps::types::TypeObjectFactory::get_instance()->build_dynamic_type(
+        test::data_type_name,
+        GetHelloWorldIdentifier(true),
+        GetHelloWorldObject(true));
+
+    TypeSupport type(new eprosima::fastrtps::types::DynamicPubSubType(test::dynamic_type_));
+    // Set type so introspection info is sent
+    type->auto_fill_type_information(true);
+    type->auto_fill_type_object(false);
+    // Register the type in the Participant
+    participant_->register_type(type);
+
+    // Create the Publisher
+    eprosima::fastdds::dds::Publisher* publisher_ = participant_->create_publisher(PUBLISHER_QOS_DEFAULT, nullptr);
+
+    // Create the DDS Topic
+    eprosima::fastdds::dds::Topic* topic_ = participant_->create_topic(topic_name, test::data_type_name,
+                    TOPIC_QOS_DEFAULT);
+
+    // Create the DDS DataWriter
+    test::writer_ = publisher_->create_datawriter(topic_, DATAWRITER_QOS_DEFAULT, nullptr);
+}
+
+eprosima::fastrtps::types::DynamicData_ptr send_sample(
+        uint32_t index)
+{
+    // Create and initialize new dynamic data
+    eprosima::fastrtps::types::DynamicData_ptr dynamic_data_;
+    dynamic_data_ = eprosima::fastrtps::types::DynamicDataFactory::get_instance()->create_data(test::dynamic_type_);
+
+    // Set index
+    dynamic_data_->set_uint32_value(index, 0);
+    // Set message
+    dynamic_data_->set_string_value(test::send_message, 1);
+    test::writer_->write(dynamic_data_.get());
+
+    usleep(100000);   // microsecond intervals - dont change
+
+    logUser(DDSRECORDER_EXECUTION, "Message published.");
```

Comment:
If showing logs, I would change this for a logInfo  so timestamps are visible. Also, please move this to before sleeping.